src/main/scala/akka/persistence/cassandra/testkit/CassandraLauncher.scala

```diff
@@ -106,24 +127,80 @@ object CassandraLauncher {
         require(cassandraDirectory.mkdirs(), s"Couldn't create Cassandra directory [$cassandraDirectory]")
 
       val realPort = if (port == 0) randomPort else port
-      val storagePort = freePort()
-
-      // http://wiki.apache.org/cassandra/StorageConfiguration
-      val conf = readResource(configResource)
-      val amendedConf = conf
-        .replace("$PORT", realPort.toString)
-        .replace("$STORAGE_PORT", storagePort.toString)
-        .replace("$DIR", cassandraDirectory.getAbsolutePath)
-      val configFile = new File(cassandraDirectory, configResource)
-      writeToFile(configFile, amendedConf)
-
-      System.setProperty("cassandra.config", "file:" + configFile.getAbsolutePath)
-      System.setProperty("cassandra-foreground", "true")
-
-      // runManaged = true to avoid System.exit
-      val daemon = new CassandraDaemon(true)
-      daemon.activate()
-      cassandraDaemon = Some(daemon)
+
+      if (fork) {
+
+        // Calculate classpath
+        val classpath = this.getClass.getClassLoader match {
+          case urlClassLoader: URLClassLoader => urlClassLoader.getURLs
+          case other                          => sys.error(s"Unable to determine classpath by inspecting CassandraLaunchers classloader, expected a URLClassLoader, but got $other")
+        }
+
+        val / = File.separator
+        val javaBin = s"${System.getProperty("java.home")}${/}bin${/}java"
+        val classpathArg = classpath.map {
+          case fileUrl if fileUrl.getProtocol.equalsIgnoreCase("file") =>
+            new File(fileUrl.toURI).getAbsolutePath
+          case other => sys.error(s"Unable to convert non file classpath URL to classpath: $other")
+        }.mkString(File.pathSeparator)
+        val className = classOf[CassandraLauncher].getCanonicalName
+
+        val builder = new ProcessBuilder(javaBin, "-cp", classpathArg, className,
+          s"$realPort", /* clean = */ "false", cassandraDirectory.getAbsolutePath, configResource)
+          .inheritIO()
+
+        val process = builder.start()
+
+        val shutdownHook = new Thread {
+          override def run(): Unit = {
+            process.destroyForcibly()
+          }
+        }
+        Runtime.getRuntime.addShutdownHook(shutdownHook)
```

Comment:
Hmm I guess this 'leaks' references to `Process` instances when running the tests from SBT right? Can we do something to avoid that (and still make sure the processes are properly stopped of course)? Perhaps we can `removeShutdownHook` them again when the test stops normally?